synapse/storage/databases/main/relations.py

```diff
@@ -132,41 +132,66 @@ def _get_recent_references_for_event_txn(
             "get_recent_references_for_event", _get_recent_references_for_event_txn
         )
 
-    async def event_has_relations(self, event_id: str) -> bool:
-        """Check if a given event has relations.
+    async def event_includes_relation(self, event_id: str) -> bool:
+        """Check if the given event includes a valid relation.
+
+        An event has a relation if it has a valid m.relates_to with a rel_type
+        and event_id in the content:
+
+        {
+            "content": {
+                "m.relates_to": {
+                    "rel_type": "m.replace",
+                    "event_id": "$other_event_id"
+                }
+            }
+        }
 
         Args:
             event_id: The event to check.
 
         Returns:
-            True if the event has any relations.
+            True if the event includes a valid relation.
         """
 
         result = await self.db_pool.simple_select_one_onecol(
             table="event_relations",
             keyvalues={"event_id": event_id},
             retcol="event_id",
             allow_none=True,
-            desc="event_has_relations",
+            desc="event_includes_relation",
         )
         return result is not None
 
-    async def event_related_to(self, parent_id: str) -> bool:
-        """Check if a given event has other event relating to it.
+    async def event_has_relation(self, parent_id: str) -> bool:
```

Comment:
definitely clunkier, but do you think `event_is_target_of_relation` might paint this better?

I think I'm beginning to see a bit more clearly, that *relations* here are really just arrows between events.
It might be sensible to call the pointer a 'source' and the pointee the 'target' (perhaps could call the other function `event_is_source_of_relation`?).
My suggestion makes sense to me here and now in this context, but it's hard to say if this would get confusing in another context or if other people were trying to make sense of it :/.

The example below is really crucial though; as for the name, I'm happy to trust what you think is best.

(An alternative to source/target would be parent/child, if that's any better — I think it makes lots of sense for reactions, threads, replies and edits so maybe that's the winner?)

Reply:
It seems that MSC2974 refers to them as "target event" and "relating event".